### PR TITLE
Port ipapython.dnssec.odsmgr to xml.etree

### PR DESCRIPTION
**Pull request by tiran**
The module ipapython.dnssec.odsmgr is the only module in ipalib,
ipaclient, ipapython and ipaplatform that uses lxml.etree.

The only dependency on lxml left in [free]ipa-client package is in the
ipa-client-automount script.

https://fedorahosted.org/freeipa/ticket/6469
https://fedorahosted.org/freeipa/ticket/6491

Signed-off-by: Christian Heimes <cheimes@redhat.com>

### CODE REVIEW

File: ipatests/.cache/v/cache/lastfailed [hunk 1]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (mbasti-rh) [inline ipatests/.cache/v/cache/lastfailed]: What is this?

File: freeipa.spec.in [hunk 2]
```diff
@@ -248,6 +248,7 @@ Requires: %{name}-server-common = %{version}-%{release}
 Requires: %{name}-common = %{version}-%{release}
 Requires: python2-ipaclient = %{version}-%{release}
 Requires: python-ldap >= 2.4.15
+Requires: python-lxml
```
Comment (mbasti-rh) [inline freeipa.spec.in]: Why did you move dependency from python-ipalib to python-ipaserver package? You are editing ipapython which is in python-ipalib package

File: ipatests/test_ipapython/test_dnssec.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
```
Comment (mbasti-rh) [inline ipatests/test_ipapython/test_dnssec.py]: Please use the new license header

File: freeipa.spec.in (hunk 2 above)
Comment (mbasti-rh) [inline freeipa.spec.in]: Why did you move dependency from python-ipalib to python-ipaserver package? You are editing ipapython module which is in python-ipalib package, so dependency should stay in python-ipalib package

**Comment by mbasti-rh** [comment]
I wrote some inline comments, I will test it later, DNSSEC is very hard to debug when an issue occurs, so review must be perfect.

File: ipatests/.cache/v/cache/lastfailed (hunk 1 above)
Comment (tiran) [inline ipatests/.cache/v/cache/lastfailed]: bah, .gitignore is missing a line to ignore .cache directories.

File: ipatests/test_ipapython/test_dnssec.py (hunk 3 above)
Comment (tiran) [inline ipatests/test_ipapython/test_dnssec.py]: Which is the new license header?

File: freeipa.spec.in (hunk 2 above)
Comment (tiran) [inline freeipa.spec.in]: Exactly, I removed the dependency on lxml from ipalib. Only ipaserver depends on lxml.

**Comment by tiran** [comment]
I just replaced the XML parser with a slightly different code. I even included unit test to ensure that the XML parser works. Feel free to add more XML files or a real world example. I couldn't find one.

File: freeipa.spec.in (hunk 2 above)
Comment (mbasti-rh) [inline freeipa.spec.in]: file `odsmgr.py` is in  python2-ipalib, so dependency python-lxml should stay with python2-ipalib not ipaserver
```
$ rpm -ql python2-ipalib | grep odsmgr
/usr/lib/python2.7/site-packages/ipapython/dnssec/odsmgr.py
/usr/lib/python2.7/site-packages/ipapython/dnssec/odsmgr.pyc
/usr/lib/python2.7/site-packages/ipapython/dnssec/odsmgr.pyo

$ # rpm -ql python2-ipaserver | grep odsmgr
$
```

You can move the whole dnssec dir/package into ipaserver with all related dependencies (in new patch) or keep python-lxml dependency with python-ipalib

Comment (tiran) [inline freeipa.spec.in]: You are misunderstanding the patch. odsmgr does **no longer** use ```python-lxml```. That's the whole point of the patch. ```xml.etree``` is part of Python's standard library.

**Comment by tiran** [comment]
client/ipa-client-automount also used ```lxml.etree```. I replaced the implementation with simpler ```xml.etree``` based code.

```
$ find . \( -type f -and -executable \) -or -name '*.py' | xargs grep lxml
./ipaserver/plugins/dogtag.py:To parse the XML documents we use the Python lxml package which is a Python
./ipaserver/plugins/dogtag.py:for many projects. One of the features in lxml and libxml2 that is particularly
./ipaserver/plugins/dogtag.py:namespaces. The regular expression name space identifier is 're:' In lxml we
./ipaserver/plugins/dogtag.py:from lxml import etree
./ipaserver/install/ipa_otptoken_import.py:from lxml import etree
./config.status:S["XMLRPC_LIBS"]="-lxmlrpc_client"
```

File: freeipa.spec.in (hunk 2 above)
Comment (mbasti-rh) [inline freeipa.spec.in]: OK

**Comment by mbasti-rh** [comment]
Commit has no ticket `Use xml.etree in ipa-client-automount script`

**Comment by mbasti-rh** [comment]
There is already DNSSEC issue that is not caused by this PR https://fedorahosted.org/freeipa/ticket/6495

**Comment by mbasti-rh** [comment]
Fixed upstream
master:
https://fedorahosted.org/freeipa/changeset/64af88fee4a482b3f393d38ff2c7f9494e689a7b
https://fedorahosted.org/freeipa/changeset/9fbd29cc106660865bc6cda225d6a8a338a78d31
